clean-abap/CleanABAP_es.md

```diff
@@ -0,0 +1,5123 @@
+Las decisiones arbitrarias hacen que el lector pierda el tiempo en la pregunta sin sentido de por qué
+se usó uno en lugar del otro, por lo tanto se debería reemplazar con decisiones precisas y bien fundamentadas.
+Nuestra recomendación está basada en el siguiente razonamiento:
+
+- Los `field-symbols` pueden hacer cosas que las referencias no pueden, como acceder dinámicamente a los componentes de una estructura.
+De la misma manera, las referencias pueden hacer cosas que los `field-symbols` no, como construir dinámicamente una estructura de datos con tipos. En resumen, elegir una sola para todas las situaciones no es posible.
+
+- En ABAP orientado a objetos, las referencias se usan por todas partes y no se pueden evitar, ya que cualquier
+objeto es una `REF TO <class-name>`. 
+En contraste, los `field-symbols` son solamente requeridos estrictamente en situaciones especiales al usar
+programación dinámica. Las referencias por lo tanto tienen una preferencia natural en cualquier programa
+orientado a objetos.
+
+- Los `field-symbols` son más cortos que las referencias, pero el ahorro resultante en memoria es tan
+pequeño que puede ser ignorado sin problema.
+De la misma manera, la velocidad no es un problema. Como consecuencia, no hay razón relacionada con el rendimiento
+para preferir uno sobre el otro.
+
+> Lee más en el artículo
+> [_Accessing Data Objects Dynamically_ in the ABAP Programming Guidelines](https://help.sap.com/doc/abapdocu_751_index_htm/7.51/en-US/index.htm?file=abendyn_access_data_obj_guidl.htm).
+
+## Tablas
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Esta sección](#tablas)
+
+### Usa el tipo correcto de tabla
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Tablas](#tablas) > [Esta sección](#usa-el-tipo-correcto-de-tabla)
+
+- Típicamente se usan tablas del tipo `HASHED` para **tablas grandes** que son 
+**llenadas en un solo paso**, **nunca se modifican** y son **leídas seguido por su llave**.
+Su overhead en memoria y procesamiento hace a las tablas hash únicamente valiosas
+cuando se tienen cantidades grandes de datos y muchos accesos de lectura.
+Cada cambio al contenido d ela tabla requiere costosos recálculos del hash,
+así que no uses este tipo para tablas que son modificadas muy seguido.
+
+- Típicamente se usan tablas del tipo `SORTED` para **tablas grandes**
+que necesitan estar **ordenadas en todo momento**, que son **llenadas poco a poco** o que 
+**necesitan ser modificadas** y **leídas por una o más llaves parciales o completas** o procesadas
+**en cierto orden**.
+Agregar, cambiar o quitar contenido require encontrar el punto de inserción adecuado,
+pero no requiere reajustar los índices de la tabla completa.
+Las tablas ordenadas demustran su valor únicamente para cantidades grandes de accesos de lectura.
+
+- Usa tablas del tipo `STANDARD` para **tablas pequeñas**, donde el indexado genera más overhead
```

Comment:
```suggestion
- Usa tablas del tipo `STANDARD` para **tablas pequeñas**, donde el indexado genera más coste
```

Reply:
Same comment on _coste_ not conveying the same meaning as overhead.